src/pem_common_plug.c

```diff
@@ -49,8 +51,13 @@ int pem_valid(char *ciphertext, struct fmt_main *self)
 	keeptr = ctcopy;
 
 	ctcopy += TAG_LENGTH;
-	if ((p = strtokm(ctcopy, "$")) == NULL) // type
+	p = strtokm(ctcopy, "$"); // type
+	if (p == NULL)
 		goto err;
+	if (*p != '1' && !warned++) {
```

Comment:
The `!warned++` trick is something we should probably move away from, because it'd fail on integer overflow (yes, very unlikely to actually trigger on valid inputs, but theoretically possible). So just `!warned` here, and `warned = 1;` in the conditional block.

Reply:
Ack